### PR TITLE
`ultralytics 8.3.232` Fix OBB NMS export lower box count

### LINKED ISSUE #22781: YOLOV11 OBB export to ONNX model losing angle or confidence
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

Hello, I have trained a obb model based on yolov11n-obb.pt. This model works perfectly fine on Ultralytics. Then I tried to export this model to onnx because I have requirement to deploy this detection on C# which only support onnx runtime.

Export code like this:
`model = YOLO("barcode_detect_v0.2.pt")

 
model.export(
    format="onnx",      
    imgsz=(640, 640),   
    opset=17,        
    nms=True,           
    device="cpu"        
)
`
And I believe obb task export was successful. I also viewed this model on netron. The output tensor is **Output shape: [1, 300, 7]**

`ONNX: starting export with onnx 1.20.0rc1 opset 17...
WARNING ⚠️ ONNX: simplifier failure: No module named 'onnxslim'
ONNX: export success ✅ 2.5s, saved as 'barcode_detect_v0.2.onnx' (10.3 MB)

Export complete (3.3s)
Results saved to /Users/admin/Desktop/individualProject/CascadeProjects/windsurf-project
Predict:         yolo predict task=obb model=barcode_detect_v0.2.onnx imgsz=640  
Validate:        yolo val task=obb model=barcode_detect_v0.2.onnx imgsz=640 data=data.yaml  
Visualize:       https://netron.app`

I have a python script of validating the result of onnx export below.

I tested many image works good on Ultralytics yolo prediction. But when I try to test this onnx result I got output like:
`Detection #1 (index=253):
  [0] x_center  =   421.2221
  [1] y_center  =    91.5810
  [2] width     =     5.9907
  [3] height    =     9.1656
  [4] angle_raw =     0.0000  (range ≈ -1 to 1)
  [5] reserved  =     0.0000  (should be 0)
  [6] logit_raw =     1.5877  → confidence = 0.830287

Detection #2 (index=252):
  [0] x_center  =   412.7953
  [1] y_center  =    90.9597
  [2] width     =     7.3966
  [3] height    =    10.4551
  [4] angle_raw =     0.0000  (range ≈ -1 to 1)
  [5] reserved  =     0.0000  (should be 0)
  [6] logit_raw =     1.5489  → confidence = 0.824750

Detection #3 (index=179):
  [0] x_center  =   421.4361
  [1] y_center  =    83.6937
  [2] width     =     6.0606
  [3] height    =     9.3021
  [4] angle_raw =     0.0000  (range ≈ -1 to 1)
  [5] reserved  =     0.0000  (should be 0)
  [6] logit_raw =     1.5182  → confidence = 0.820271

Detection #4 (index=178):
  [0] x_center  =   413.1988
  [1] y_center  =    83.4118
  [2] width     =     6.6288
  [3] height    =    10.8000
  [4] angle_raw =     0.0000  (range ≈ -1 to 1)
  [5] reserved  =     0.0000  (should be 0)
  [6] logit_raw =     1.4693  → confidence = 0.812944
....
`

Almost every detection dont have none zero value on index 4and 5. Since I try to make the last digit to be confidence score I can finally got some reasonable result after filtering. Can someone explain why this is wired or how I should postprocess the onnx prediction result?
I have checked past github issues but still no similar content.

### Environment

ltralytics 8.3.228 🚀 Python-3.14.0 torch-2.9.1 CPU (Apple M4)
Setup complete ✅ (10 CPUs, 16.0 GB RAM, 111.9/460.4 GB disk)

OS                     macOS-15.3.1-arm64-arm-64bit-Mach-O
Environment            Darwin
Python                 3.14.0
Install                pip
Path                   /Users/admin/Desktop/individualProject/CascadeProjects/windsurf-project/.venv/lib/python3.14/site-packages/ultralytics
RAM                    16.00 GB
Disk                   111.9/460.4 GB
CPU                    Apple M4
CPU count              10
GPU                    None
GPU count              None
CUDA                   None

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.7>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 12.0.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.16.3>=1.4.1
torch                  ✅ 2.9.1>=1.8.0
torch                  ✅ 2.9.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.24.1>=0.9.0
psutil                 ✅ 7.1.3
polars                 ✅ 1.35.2
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

import cv2
import numpy as np
import onnxruntime as ort
import sys

def sigmoid(x):
    return 1.0 / (1.0 + np.exp(-x))

def preprocess_image(image, target_size=640):
    h, w = image.shape[:2]
    
    scale = min(target_size / w, target_size / h)
    
    new_w = int(w * scale)
    new_h = int(h * scale)
    
    img_resized = cv2.resize(image, (new_w, new_h), interpolation=cv2.INTER_LINEAR)
    
    pad_w = (target_size - new_w) // 2
    pad_h = (target_size - new_h) // 2
    
    img_padded = np.full((target_size, target_size, 3), 114, dtype=np.uint8)
    img_padded[pad_h:pad_h+new_h, pad_w:pad_w+new_w] = img_resized
    
    img_rgb = cv2.cvtColor(img_padded, cv2.COLOR_BGR2RGB)
    
    img_norm = img_rgb.astype(np.float32) / 255.0
    
    img_chw = np.transpose(img_norm, (2, 0, 1))
    
    img_batch = np.expand_dims(img_chw, axis=0)
    
    return img_batch, scale, pad_w, pad_h

def main():
    if len(sys.argv) < 2:
        print("Usage: python inspect_all_channels.py <image_path> [model_path]")
        print("Example: python inspect_all_channels.py test_image.jpg")
        print("         python inspect_all_channels.py test_image.jpg barcode_detect_v0.2.onnx")
        sys.exit(1)
    
    image_path = sys.argv[1]
    model_path = sys.argv[2] if len(sys.argv) > 2 else "barcode_detect_v0.2.onnx"
    
    print("="*80)
    print("ONNX Model Channel Inspector")
    print("="*80)
    print(f"Image path: {image_path}")
    print(f"Model path: {model_path}")
    
    print("\nLoading model...")
    session = ort.InferenceSession(model_path, providers=['CPUExecutionProvider'])
    
    input_info = session.get_inputs()[0]
    output_info = session.get_outputs()[0]
    
    print(f"✓ Model loaded successfully")
    print(f"  Input shape: {input_info.shape}")
    print(f"  Output shape: {output_info.shape}")
    
    print(f"\nReading image...")
    image = cv2.imread(image_path)
    if image is None:
        print(f"✗ Failed to read image: {image_path}")
        sys.exit(1)
    
    print(f"✓ Image loaded successfully")
    print(f"  Original size: {image.shape[1]}x{image.shape[0]}")
    
    print(f"\nPreprocessing image...")
    input_tensor, scale, pad_w, pad_h = preprocess_image(image)
    print(f"✓ Preprocessing completed")
    print(f"  Scale ratio: {scale:.4f}")
    print(f"  Padding: ({pad_w}, {pad_h})")
    
    print(f"\nRunning inference...")
    input_name = session.get_inputs()[0].name
    output_name = session.get_outputs()[0].name
    
    outputs = session.run([output_name], {input_name: input_tensor})
    output = outputs[0]
    
    print(f"✓ Inference completed")
    print(f"  Output shape: {output.shape}")
    
    print("\n" + "="*80)
    print("Output Data Analysis")
    print("="*80)
    
    if len(output.shape) == 3:
        batch, num_detections, num_channels = output.shape
        print(f"Batch size: {batch}")
        print(f"Number of detections: {num_detections}")
        print(f"Number of channels: {num_channels}")
        
        channel_names = [
            "ch0",
            "ch1", 
            "ch2",
            "ch3",
            "ch4",
            "ch5",
            "ch6"
        ]
        
        ch6_sigmoid = sigmoid(output[0, :, 6])
        top_indices = np.argsort(ch6_sigmoid)[::-1][:10]
        
        print("\n" + "="*80)
        print("Top 10 Detections (sorted by ch6 sigmoid confidence)")
        print("="*80)
        
        for rank, idx in enumerate(top_indices, 1):
            conf = ch6_sigmoid[idx]
            print(f"\nDetection #{rank} (index={idx}, confidence={conf:.6f}):")
            for ch in range(num_channels):
                val = output[0, idx, ch]
                if ch == 6:
                    print(f"  [{ch}] = {val:10.6f}  (sigmoid: {conf:.6f})")
                else:
                    print(f"  [{ch}] = {val:10.6f}")
        
        output_file = image_path.replace('.jpg', '_channels.npy').replace('.png', '_channels.npy')
        np.save(output_file, output)
        print("\n" + "="*80)
        print(f"✓ Full data saved to: {output_file}")
        print(f"  Load with: np.load('{output_file}')")
        print("="*80)
        
    else:
        print(f"⚠️  Output shape is not 3D: {output.shape}")
        print("Raw output:")
        print(output)

if __name__ == "__main__":
    main()

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes #22781 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Small release bump that tweaks NMS box normalization logic for exported models and adds a new docs author entry. 🚀

---

### 📊 Key Changes
- 🔢 **Version bump**: Updated package version from `8.3.231` to `8.3.232`.
- 👤 **Docs authors update**: Added GitHub user `@akdanz` (avatar and username) to `mkdocs_github_authors.yaml` for documentation attribution.
- 📦 **Exporter NMS tweak**: Adjusted the `multiplier` used in the exporter’s NMS box normalization:
  - For OBB (`self.obb == True`): `multiplier` is now fixed at `8`.
  - For non-OBB (`self.obb == False`): `multiplier` is now `1 / max(len(self.model.names), 1)`.

---

### 🎯 Purpose & Impact
- 📐 **More controlled NMS behavior**:  
  - Keeps a strong offset (`8`) for oriented bounding boxes, which can help preserve correct NMS behavior in OBB scenarios.
  - Scales boxes more conservatively for non-OBB models by factoring in the number of classes.
- 🤖 **Better export robustness**:  
  - The normalization change is specifically relevant for formats like TFLite, where large class offsets and int8 quantization can cause issues. This tweak aims to improve stability and correctness of exported models.
- 🙌 **Improved contributor recognition**:  
  - Ensures `@akdanz` is properly credited in the docs, supporting clearer authorship and community contributions.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated review assistant and a Ultralytics engineer will be with you soon. In the meantime, please review the checklist below to help streamline the process:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Thanks for the update—after reviewing the change, I didn’t spot any issues. The OBB multiplier adjustment looks correct and keeps non-OBB behavior unchanged.

**Comment by UltralyticsAssistant** [comment]
What a fantastic merge, @Y-T-G — thank you for this thoughtful refinement, and kudos to @glenn-jocher for the collaborative touches behind it. 🚀  

> “Details create the big picture.” – Sanford I. Weill  

This PR is a perfect example of that: a precise NMS normalization tweak that quietly boosts export robustness (especially for quantized formats like TFLite), a clean version bump to `8.3.232`, and improved recognition for contributors like @akdanz in our docs. These may look small in isolation, but together they push the ecosystem forward and make life easier for everyone building with Ultralytics.  

Really appreciate the care and intention behind this work — contributions like this keep the project sharp and the community strong. 🙌
